src/components/dragLineLayer.ts

```diff
@@ -0,0 +1,170 @@
+    private _dragInteraction: Interactions.Drag;
+    private _detectionRadius = 3;
+    private _detectionEdge: d3.Selection<void>;
+    private _enabled = true;
+
+    private _dragStartCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _dragEndCallbacks: Utils.CallbackSet<DragLineCallback<D>>;
+    private _disconnectInteractionCallbacks: () => void;
+
+    constructor(orientation: string) {
+      super(orientation);
+      this.addClass("drag-line-layer");
+      this.addClass("enabled");
+      if (this._isVertical()) {
+        this.addClass("vertical");
+      } else {
+        this.addClass("horizontal");
+      }
+
+      this._dragInteraction = new Plottable.Interactions.Drag();
+      this._dragInteraction.attachTo(this);
+
+      let grabbedLine = (p: Point) => {
+        return (this._isVertical() &&
+                 p.x >= this.pixelPosition() - this.detectionRadius() &&
+                 p.x <= this.pixelPosition() + this.detectionRadius()
+               ) ||
+               (
+                 !this._isVertical() &&
+                 p.y >= this.pixelPosition() - this.detectionRadius() &&
+                 p.y <= this.pixelPosition() + this.detectionRadius()
+               );
+      };
+
+      let dragging = false;
+      let dragStartCallback = (start: Point) => {
+        if (grabbedLine(start)) {
+          dragging = true;
+          this._dragStartCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragStart(dragStartCallback);
+      let dragCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          this._setPixelPositionWithoutChangingMode(this._isVertical() ? end.x : end.y);
+          this._dragCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDrag(dragCallback);
+      let dragEndCallback = (start: Point, end: Point) => {
+        if (dragging) {
+          dragging = false;
+          this._dragEndCallbacks.callCallbacks(this);
+        }
+      };
+      this._dragInteraction.onDragEnd(dragEndCallback);
+
+      this._disconnectInteractionCallbacks = () => {
+        this._dragInteraction.offDragStart(dragStartCallback);
+        this._dragInteraction.offDrag(dragCallback);
+        this._dragInteraction.offDragEnd(dragEndCallback);
+        delete this._disconnectInteractionCallbacks;
+      };
+
+      this._dragStartCallbacks = new Utils.CallbackSet<DragLineCallback<D>>();
+      this._dragCallbacks = new Utils.CallbackSet<DragLineCallback<D>>();
+      this._dragEndCallbacks = new Utils.CallbackSet<DragLineCallback<D>>();
+    }
+
+    protected _setup() {
+      super._setup();
+      this._detectionEdge = this.content().append("line").style({
+                              opacity: 0,
+                              fill: "pink",
+                              "pointer-events": "visibleStroke"
```

Comment:
Interesting. This is mostly for the benefit of non-IE browsers that care about it (Safari specifically) -- `DragBoxLayer` does this too.